.github/workflows/tests.yml

```diff
@@ -24,15 +25,16 @@ jobs:
     - name: Install required packages
       run: |
         sudo apt-get update
-        sudo apt-get install libcurl4-openssl-dev libsystemd-dev libjson-glib-dev
+        sudo apt-get install meson libcurl4-openssl-dev libsystemd-dev libjson-glib-dev
 
-    - name: Build (with ${{ matrix.options }})
+    - name: Build (with ${{ matrix.options }} ${{ matrix.flags }})
       run: |
-        mkdir -p build
-        cd build
-        cmake ${{ matrix.options }} -DQA_BUILD=ON ..
-        make
-        cd -
+        ${{ matrix.flags }} meson meson_build ${{ matrix.options }} -Dwerror=true
```

Comment:
`QA_BUILD` passed more options than `-Werror`. What about them?

Reply:
I've defined `warning_level=3` in the `default_options` which should enable the rest like `-Wextra` `-pedantic`, etc and '-Dwerror` is set explicitly in test